apps/calendar/js/views/modify_event.js

```diff
@@ -6,15 +6,18 @@ var EventBase = require('./event_base');
 var InputParser = require('shared/input_parser');
 var Local = require('provider/local');
 var QueryString = require('querystring');
+var calendarObserver = require('calendar_observer');
 var dateFormat = require('date_format');
+var forEach = require('object').forEach;
 var getTimeL10nLabel = require('calc').getTimeL10nLabel;
-var nextTick = require('next_tick');
 
 require('css!modify_event_view');
 require('dom!modify-event-view');
 
 function ModifyEvent(options) {
+  this._loadingCalendarList = true;
```

Comment:
Add a flag to indicate whether or not we've received a change notification from `calendarObserver`